api/src/routes/crud.ts

```diff
@@ -501,21 +501,91 @@ router.post("/medication-request", [requireJWTMiddleware], async (req: Request,
                 }
             
             } else {
-                console.log(`Error: property ${medication} not found`);                
+                console.log(`Error: property ${medication} not found`);
             }
 
             const { siteCoding, routeCoding, methodCoding, unitCoding } = administrationCodings;
-            const medicationRequestDetails = MedicationRequest({ id, ...req.body[medication] }, medicationCoding, siteCoding, routeCoding, methodCoding, unitCoding);    
+            const medicationRequestDetails = MedicationRequest({ id, ...req.body[medication] }, medicationCoding, siteCoding, routeCoding, methodCoding, unitCoding);
             
-            await FhirApi({ url: `/MedicationRequest/${id}`, method: "PUT", data: JSON.stringify(medicationRequestDetails) });              
+            await FhirApi({ url: `/MedicationRequest/${id}`, method: "PUT", data: JSON.stringify(medicationRequestDetails) });
             
             medicationRequestIds.push({ medication, status: "success", medicationRequestId: id });
-        }     
+        }
         
-        res.json({ status: "success", data: medicationRequestIds });                    
+        res.json({ status: "success", data: medicationRequestIds });
     } catch (err) {
         res.json({ err, status: "error" });
     }
-})
+});
+
+router.post("/allergy-intolerance", [requireJWTMiddleware], async (req: Request, res: Response) => {
+    
+    try {
+
+        let id = uuidv4();
+
+        const { patientId, vaccine } = req.body;
+
+        const getVaccineCoding = () => {
+
+            if (Object.keys(medicalCodes).indexOf(vaccine) > -1) {            
+                return ({ status: "success", data: medicalCodes[vaccine] });                
+            } else {
+                return ({status:"error", data:`Error: property ${vaccine} not found`});
+            }
+        }
+
+        const { status, data } = getVaccineCoding();
+
+        if (status !== "success") {
+            res.statusCode = 400;
+            res.json(data);            
+            return;
+        }
+
+        const vaccineCode = data.split(":")[1];
```

Comment:
Is there a fallback mechanism if medicalCodes[vaccine] is not found?

Reply:
If medicalCodes[vacine] is not found, we inform the user that the vaccine they want to submit as the cause of the adverse reaction is not found within the system. The logic to check the existence of the vaccine is handled here: `if (Object.keys(medicalCodes).indexOf(vaccine) > -1)` and if the vaccine is not found, its handled in the else statement. This info is relayed to the user here: `        if (status !== "success") {
            res.statusCode = 400;
            res.json(data);            
            return;
        }`
The vaccine coding has to be present in the medicalCodes object. If not we cannot pass the vaccine coding to the allergy intolerance, hence we terminate the request, as seen above with the return statement. Important to note, however, is that the vaccines are finite and so all those used in the kenyan healthcare system are included in the medicalCodes object and the client has to pass a vaccine name corresponding to its key in the medicalCodes object key.